qiskit_nature/circuit/library/ansatzes/utils/fermionic_excitation_generator.py

```diff
@@ -106,6 +96,19 @@ def generate_fermionic_excitations(
     """Generates all possible excitations with the given number of excitations for the specified
     number of particles distributed among the given number of spin orbitals.
 
+    The method must be called for each type of excitation (singles, doubles, etc.) that is to be
+    considered in the Ansatz. Some keyword arguments are allowed through which the excitations can
+    be filtered to fit the user's needs. `alpha_spin` and `beta_spin` are boolean flags which can
+    be set to `False` in order to disable the inclusion of alpha-spin or beta-spin excitation,
+    respectively. `max_spin_excitation` takes an integer value which defines the maximum number of
+    excitations that can occur within the same spin. Thus, setting `max_spin_excitation=1` and
+    `num_excitations=2` yields only those double excitations which do not excite the same spin species
+    twice. `generalized` is another boolean flag which enables generalized excitations which are
+    effectively ignoring the spin orbital occupancies. Excitations will be produced based on an
+    initial `Hartree-Fock` occupation by default unless `generalised` is set to `True`. Therefore,
+    the excitations are only determined based on the number of spin orbitals and are independent from
+    the number of particles.
+
```

Comment:
An overall comment. This seems to be a bit of a duplication of text that is Args. I am wondering, if further explanation is needed (which seems to be the case here), perhaps improve the args text a bit, and also if it relates to values of args interacting maybe just having some example calls here with explanation whats happening and the corresponding output might be more helpful, easier to understand? These can be done as doctest so they would be examples that will be checked by CI which verifies the output matches that stored in the example when run.

For this text `The method must be called for each type of excitation (singles, doubles, etc.) that is to be considered in the Ansatz.` I think it might be better to just end it at considered (ie. drop in the Ansatz). In the past in Aqua we used a similar excitation list generator in places other than an ansatz where it applied (e.g. as part of the computation of mp2 perturbation coeffs). I think this here is more of a common utility even though its nested under the ansatzes.

Reply:
Yep I agree with @woodsp-ibm here.

For some context: I had discussed with @javabster offline that we should check the contents of the top-level docstring in this file for information which should be moved/copied into this methods docstring. The reason being that the top-level docstring is not actually included in the documentation because only this specific method is exported in the `__init__` of this module. If you think that this ought to be changed, @woodsp-ibm, please let me know. I did not see a reason for exposing `get_{alpha|beta}_excitations` so I tend to say that this is not needed.

That being said, @javabster if you could try to deduplicate the information in this docstring and maybe add some examples, that would be great. You can see an example of how an example (:sweat_smile:) is added to a docstring here: https://github.com/Qiskit/qiskit-nature/blob/be91c8e9657082d8670fd16fd79d02b565fa9f9c/qiskit_nature/operators/second_quantization/fermionic_op.py#L157-L166